### PR TITLE
529 set dark mode in user preferences

### PR DESCRIPTION
**Pull request by dkrakau**
### Description

ISSUE: #529  set dark mode in user preferences

## Checklist

- [ ] My code follows the style guidelines
- [ ] I have documented my code
- [ ] My changes generate no new warnings
- [ ] I have linked the Issue
- [ ] The Pull Request covers all the Checkpoints from the tagged Issue
- [ ] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [x] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [ ] No tests required
- [x] Yes
- [ ] No

### CODE REVIEW

File: frontend/src/components/map/LeafletMapContainer.tsx [hunk 1]
```diff
@@ -45,7 +46,7 @@ const LeafletMapContainer = ({ originId, destinationId, locationIds, showLines,
       polygons.push(
         <Polygon
           key={"line" + i}
-          color={window.matchMedia("(prefers-color-scheme: dark)") ? "rgb(0,148,236)" : "rgb(77, 77, 77)"}
+          color={"rgb(0,148,236)"}
```
Comment (LaraBiss) [inline frontend/src/components/map/LeafletMapContainer.tsx:49]: For what is this color used? In general no color should be hardcoded maybe there is an opportunity to use a color from our color scheme.

File: frontend/src/components/map/ReactiveMapContainer.tsx [hunk 2]
```diff
@@ -47,7 +47,7 @@ const ReactiveMapContainer: React.FC<PropsWithChildren<ReactiveContainerProps>>
 
     const view = props.view;
     if (isCenterZoomView(view)) {
-      //setTimeout(() => map.flyTo(view.center, view.zoom, { animate: true, duration: 0.5 }), 250);
+      //setTimeout(() => map.flyTo(view.center, view.zoom, { animate: true, duration: 0.5 }), 50);
```
Comment (LaraBiss) [inline frontend/src/components/map/ReactiveMapContainer.tsx:50]: Is having the comment necessary?

File: frontend/src/pages/UserPreferencesPage.tsx [hunk 3]
```diff
@@ -1,23 +1,75 @@
...
+        : props.setIsDarkThemeEnabled(false)
+      : props.setIsDarkThemeEnabled(false);
+    props.isDarkThemeEnabled
+      ? document.body.setAttribute("color-theme", "dark")
+      : document.body.removeAttribute("color-theme");
+  }, [props.isDarkThemeEnabled]);
+
+  return (
+    <IonPage id="main-content">
+      <IonHeader>
```
Comment (LaraBiss) [inline frontend/src/pages/UserPreferencesPage.tsx:47]: Please don't use the IonHeader to make your own header. There is a Header Component for consistency.

File: frontend/src/theme/variables.css [hunk 4]
```diff
@@ -85,164 +85,165 @@ http://ionicframework.com/docs/theming/ */
   --ion-color-light-tint: #f5f6f9;
 }
 
-@media (prefers-color-scheme: dark) {
-  /*
+/* @media (prefers-color-scheme: dark) { */
```
Comment (LaraBiss) [inline frontend/src/theme/variables.css:88]: Comment

File: frontend/src/theme/variables.css [hunk 5]
```diff
@@ -85,164 +85,165 @@ http://ionicframework.com/docs/theming/ */
...
+  --ion-tab-bar-background: #1f1f1f;
+
+  --ion-card-background: #1e1e1e;
 }
+
+.dark-map {
+  filter: var(--dark-map-filter, none);
 }
+
+/* } */
```
Comment (LaraBiss) [inline frontend/src/theme/variables.css:249]: Comment

**Review by LaraBiss — CHANGES_REQUESTED** [review]

File: frontend/src/components/map/LeafletMapContainer.tsx (hunk 1 above)
Comment (dkrakau) [reply frontend/src/components/map/LeafletMapContainer.tsx:49]: Polygon is a map component from Leafletmap and i don't think there is an other way to set the dashed line color.

File: frontend/src/components/map/ReactiveMapContainer.tsx (hunk 2 above)
Comment (dkrakau) [reply frontend/src/components/map/ReactiveMapContainer.tsx:50]: I have deleted the comments

File: frontend/src/pages/UserPreferencesPage.tsx (hunk 3 above)
Comment (dkrakau) [reply frontend/src/pages/UserPreferencesPage.tsx:47]: Header component is used instead.

File: frontend/src/theme/variables.css (hunk 4 above)
Comment (dkrakau) [reply frontend/src/theme/variables.css:88]: I have deleted the comments

File: frontend/src/theme/variables.css (hunk 5 above)
Comment (dkrakau) [reply frontend/src/theme/variables.css:249]: I have deleted the comments

**Review by LaraBiss — APPROVED** [review]

**Pull request merged**